pkg/stats/monitor.go

```diff
@@ -203,8 +222,30 @@ func (m *Monitor) updateEgressStats(idle float64, usage map[int]float64) {
 
 		if procStats.egressID != "" {
 			m.promProcCPULoad.With(prometheus.Labels{"egress_id": procStats.egressID}).Set(cpuUsage)
+			if cpuUsage > procStats.allowedUsage && cpuUsage > maxUsage {
+				maxUsage = cpuUsage
+				maxEgress = procStats.egressID
+			}
 		}
 	}
+
+	if load > m.killThreshold {
+		logger.Warnw("high cpu usage", nil,
+			"load", load,
+			"requests", m.requests.Load(),
+		)
+
+		if m.requests.Load() > 1 {
+			if m.highCPUCount < 3 {
+				m.highCPUCount++
+				return
+			} else {
+				m.killProcess(maxEgress)
```

Comment:
maxEgress could be empty? I guess that might be fine.

Reply:
I couldn't think of a scenario where that'd be possible, but it'll be a no-op if it happens